plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -127,6 +127,14 @@ Credential createCredentialHelper(String accessToken, String refreshToken) {
     return credential;
   }
 
+  public void clearCredential() {
+    MApplication application = PlatformUI.getWorkbench().getService(MApplication.class);
+
+    Map<String, Object> synchronizedMap =
```

Comment:
Hmm, using a synchronizeMap doesn't do anything here — other accessors won't be going through your map, and other callers to this method will get a new synchronizedMap.

If you're really concerned, put an `AtomicReference`